email/ncmec_test.go

```diff
@@ -0,0 +1,163 @@
+package email
+
+import (
+	"strings"
+	"testing"
+	"time"
+)
+
+// TestNCMECClient contains a couple of unit tests that verify the functionality
+// of the NCMEC client.
+func TestNCMECClient(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	// os.Setenv("NCMEC_USERNAME", "[ENTER_USERNAME]")
+	// os.Setenv("NCMEC_PASSWORD", "[ENTER_PASSWORD]")
+	// os.Setenv("NCMEC_DEBUG", "true")
+
+	// load credentials from env
+	creds, err := LoadNCMECCredentials()
+	if err != nil {
+		t.Log("NCMEC credentials not found, skipping...")
+		t.SkipNow()
+	}
+
+	// ensure it's a debug client
+	if !creds.Debug {
+		t.Fatal("never run in production mode")
+	}
+
+	// instantiate the client
+	client := NewNCMECClient(creds)
+
+	tests := []struct {
+		name string
+		test func(t *testing.T)
+	}{
+		{
+			name: "finishReport",
+			test: func(t *testing.T) { testFinishReport(t, client) },
+		},
+		{
+			name: "openReport",
+			test: func(t *testing.T) { testOpenReport(t, client) },
+		},
+		{
+			name: "status",
+			test: func(t *testing.T) { testStatus(t, client) },
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, test.test)
+	}
+}
+
+// testFinishReport is a unit test that verifies whether we can finish a report
+func testFinishReport(t *testing.T, c *NCMECClient) {
+	// open a report
+	now := time.Now().Add(-time.Hour)
+	report := newTestReport(now)
+	res, err := c.openReport(report)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// finish the report
+	ress, err := c.finishReport(res.ReportId)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// assert the report was finished
+	if ress.ResponseCode != 0 {
+		t.Fatalf("unexpected response code, %v != 0", ress.ResponseCode)
+	}
+	if ress.ReportId != res.ReportId {
+		t.Fatalf("unexpected report id, %v != %v", ress.ReportId, res.ReportId)
+	}
+}
+
+// testOpenReport is a unit test that verifies we can open a report with NCMEC
+func testOpenReport(t *testing.T, c *NCMECClient) {
+	// open a report
+	now := time.Now().Add(time.Hour)
+	report := newTestReport(now)
+	res, err := c.openReport(report)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// assert the response, we expect it to have failed because the invalid
+	// (future) date
+	if res.ResponseCode != 4100 {
+		t.Fatalf("unexpected response code, %v != 4100", res.ResponseCode)
+	}
+	if !strings.Contains(res.ResponseDescription, "must be a past date") {
+		t.Fatalf("unexpected response description '%s'", res.ResponseDescription)
+	}
+	if res.ReportId != 0 {
+		t.Fatalf("unexpected report id, %v", res.ReportId)
+	}
+
+	// open a report in the past
+	now = time.Now().Add(-time.Hour)
```

Comment:
We have a mix of UTC and non-UTC times in the code. We should stick to UTC.

![Screenshot 2022-03-25 at 11 18 14](https://user-images.githubusercontent.com/1464871/160102260-94db2036-2671-43ca-88d1-2e43f9f11cbb.png)

Reply:
Oh wow that's a lot of calls. Maybe we can add a helper method within the relevant package.

Reply:
I don't think that would be an improvement, tbh. When coding I would still type `time.Now().UTC()`, it's in my muscle memory. Then you'll ask me to use the helper in the code review... Just more time wasted?
Your call but I don't see that as an improvement.

Reply:
I'll update it to UTC everywhere, usually I just don't care in unit tests, it's an extra couple of characters to type and it's longer.

Reply:
Yes, it's that important in unit tests but it does help.
I was more worried about `finalizer.go:127`.

Reply:
I'll resolve this, I agree with Ivo on the helper. This looks like a ton of occurrences but it's 70% testing and I think it's better to explicitly use `.UTC()`